pyctbgui/services/Transceiver.py

```diff
@@ -69,6 +71,76 @@ def updateLegend(self):
             for plot, name in self.getEnabledPlots():
                 self.legend.addItem(plot, name)
 
+    def processWaveformData(self, data, dsamples, tsamples):
```

Comment:
Would it be possible to have this as a free function that we can test with fake data?